### PR TITLE
Call /ping periodically to track domain activity.

### PR DESCRIPTION
**Pull request by mrstegeman**
@andrenatal Can you update the config file on the server to evict old registrations every 12 or 24 hours? Not sure what it's presently set to.

### CODE REVIEW

**Comment by benfrancis** [comment]
@mrstegeman What problem is this solving?

Tip: It would be helpful if you could file issues in GitHub (bugs, tasks and user stories for features) for any issues you find and have pull requests reference the issue they're closing so that we can track and prioritise work that needs doing.

The setTimeout in this pull request raised a red flag for me, but I don't understand what problem it's trying to solve. Why is there a need to check the local IP address every hour?

Thanks

**Comment by mrstegeman** [comment]
According to the registration server API document, we are supposed to call /register periodically in order to keep the route to the gateway up to date.
https://github.com/mozilla-iot/registration_server/blob/master/api.md

Currently, we're never calling this API. On the bright side, the registration server only evicts "old" entries on startup. However, my plan is to evict them on a more regular basis. As I start to implement the domain recovery flow, it will be useful to know which domains are actively being used, to avoid stomping on active domains.

As far as the every hour part, it's entirely possible that a home router will change the gateway's DHCP reservation at some point while it's running.

**Comment by andrenatal** [comment]
@mrstegeman I created an account for you in the AWS panel, so please feel free to deploy any change that might be necessary.

**Comment by benfrancis** [comment]
> According to the registration server API document, we are supposed to call /register periodically in order to keep the route to the gateway up to date.
> https://github.com/mozilla-iot/registration_server/blob/master/api.md

Ugh, reading that document reminds me how un-RESTful this "REST API" is.

`The REST api follows these general rules... All the requests are GET requests.`

I don't think I'm being particularly dogmatic when I say this misses the point about what a REST API is! If someone wanted to take over responsibility for this API and design a better one, that would be great!

Anyway...

`
The server will evict old entries on a regular basis so the gateway needs to register itself often enough.`

This is vague... What does it actually do? My gateway uses DHCP and I've never noticed it stop working because the gateway doesn't call this API method. Does anyone know how it actually uses the local IP address and why it needs sending to the cloud server from the gateway? I'm afraid my Rust is a bit rusty.

If this is necessary, is there a better way to schedule regular tasks in NodeJS than a setTimeout?

**Comment by mrstegeman** [comment]
The registration server keeps a record of local and public IP address mappings for each gateway. Periodically, those entries are "evicted", meaning that all the gateway and domain data still exists, but the IP mappings (which are used for responding to DNS queries and such) are wiped.

Given that we were never calling the API and things still work fine, I'm not entirely sure why it exists. I'm guessing it works fine because things are just routed through PageKite tunnels. However, I was hoping to use this info for figuring out active/inactive domains for domain recovery purposes, and given that there was already an API that provided data I wanted, I figured it would be best to go ahead and use that.

If you'd prefer I pare down the APIs instead, and only use what we actually need, I can do that. Otherwise, I can look into an alternative to setTimeout().

**Comment by benfrancis** [comment]
We inherited the registration server code from a previous project so I think it's worth getting a better understanding of how it works and moulding it more to our needs if necessary. I'll see if Fabrice can help explain the original design a bit more on IRC.

In the meantime, you said you wanted to use this to identify "active/inactive domains for domain recovery purposes". Can you explain this thinking a bit more in #379?

**Comment by mrstegeman** [comment]
https://github.com/mozilla-iot/gateway/issues/379#issuecomment-342478155

**Comment by mrstegeman** [comment]
This goes along with the following registration server PR:
https://github.com/mozilla-iot/registration_server/pull/42

**Comment by mrstegeman** [comment]
The new /ping API is now live.

File: src/ssltunnel.js [hunk 1]
```diff
@@ -87,15 +92,32 @@ var TunnelService = {
...
             return true;
         }
 
         return false;
-    }
+    },
+
+    // method to ping the registration server to track active domains
+    pingRegistrationServer: function() {
+        Settings.get('tunneltoken').then((result) => {
```
Comment (benfrancis) [inline src/ssltunnel.js]: One minor point. Is it necessary to do this I/O every hour and wear out the SD card? :P Can't we store the token in memory and update it when it actually changes (which currently as far as I know it doesn't!).

Comment (mrstegeman) [reply src/ssltunnel.js]: Done.

**Pull request merged**